helios-client/src/main/java/com/spotify/helios/common/descriptors/RollingOperationStatus.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.common.descriptors;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * The state of a rolling operation.
+ *
+ * An sample expression of it in JSON might be:
+ * <pre>
+ * {
+ *   "state":"FAILED",
+ *   "error":"Stopped by user"
+ * }
+ * </pre>
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class RollingOperationStatus extends Descriptor {
```

Comment:
We're not very consistent with this at the moment. My implementation follows what pattern I can see - i.e. this isn't used in `DeploymentGroupStatus` either (but is in `DeploymentGroup`).

FWIW I'm not a fan of the concept in general. I prefer a JSON API with `nil` values than a JSON API that returns different keys under different circumstances.